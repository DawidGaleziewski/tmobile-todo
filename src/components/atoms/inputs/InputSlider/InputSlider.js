import React from "react";

// Libraries
import { css, jsx } from "@emotion/core";

// Styles
import {
  inputWrapperStyle,
  inputStyle,
  labelStyle,
} from "./style/InputSliderStyle";

/** @jsx jsx */

const InputSlider = ({ id, type, name, labelText, value, onChange }) => {
  return (
    <div
      className="input-slider-wrapper"
      css={inputWrapperStyle}
      data-testid="input-slider"
    >
      <input
        onChange={onChange}
        css={inputStyle}
        id={id}
        type={type}
        name={name}
        value={value}
        placeholder=" "
      />
      <label css={labelStyle} htmlFor={id}>
        {labelText}
      </label>
    </div>
  );
};

export default InputSlider;
