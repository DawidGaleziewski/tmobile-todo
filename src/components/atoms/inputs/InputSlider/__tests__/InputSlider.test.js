// Libraries
import React from "react";
import ReactDOM from "react-dom";
import { render, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";

// Components
import InputSlider from "../InputSlider";

afterEach(cleanup);

it("renders InputSlider without crashing", () => {
  const divContainer = document.createElement("div");

  ReactDOM.render(<InputSlider />, divContainer);
  ReactDOM.unmountComponentAtNode(divContainer);
});

it("renders input with correct text", () => {
  const { getByTestId } = render(<InputSlider labelText="lebel test text" />);
  expect(getByTestId("input-slider")).toHaveTextContent("lebel test text");
});

it("matches snapshot 1", () => {
  const DOMTree = renderer.create(<InputSlider />).toJSON();

  expect(DOMTree).toMatchSnapshot();
});
