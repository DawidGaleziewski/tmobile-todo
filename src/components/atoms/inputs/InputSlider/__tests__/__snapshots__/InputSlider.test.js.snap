// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1 1`] = `
<div
  className="input-slider-wrapper css-1lncui6-InputSlider"
  data-testid="input-slider"
>
  <input
    className="css-1wvyecb-InputSlider"
    placeholder=" "
  />
  <label
    className="css-f515j6-InputSlider"
  />
</div>
`;
